Sources/KSCrashRecording/KSCrashAppMemoryTracker.m

```diff
@@ -75,6 +123,73 @@ - (void)dealloc
     [self stop];
 }
 
+// Observers are either an object passed in that
+// implements `KSCrashAppMemoryTrackerObserving` or a block.
+// Both will be wrapped in a `KSCrashAppMemoryTrackerBlockObserver`.
+// if a block, then it'll simply call the block.
+// If the object, we'll keep a weak reference to it.
+// Objects will be reaped when their block and their object
+// is nil.
+// We'll reap on add and removal or any type of observer.
+- (void)_locked_reapObserversOrObject:(id)object
```

Comment:
[from](https://github.com/kstenerud/KSCrash/pull/678#discussion_r2278443208) @kstenerud:

"If we set up the API to only accept blocks, and then stored the blocks in an [NSPointerArray](https://www.infragistics.com/blogs/nspointerarray/), we'd get reaping for free (although your add routine would need to look for holes, but that's easy enough).

Then the user must save the reference to their block so that it doesn't go poof (our add method could return the block they passed in to make saving it more ergonomic), and we merely keep a weak ref to the block rather than copying it.

Then when they want to de-register, they simply nil out their reference to the block, and everything falls into place like magic."

Reply:
@kstenerud That's also a pretty good way to take the problem on. I Like what we did here (same as in AppState as well) because it allows you to register and forget which is what works for most users of this service. I'd keep it as is if it's all the same to you.

Reply:
done!